internal/lang/eval_test.go

```diff
@@ -421,6 +421,47 @@ func TestScopeEvalContext(t *testing.T) {
 	}
 }
 
+// TestScopeEvalContextWithParent tests if the resulting EvalCtx has correct parent.
```

Comment:
nit: Technically the tests here probably should check that the EvalContext still contains the necessary vars and functions

Reply:
I agree, I extended the test with value-based checks for Variables and Functions: [fa3d716](https://github.com/opentofu/opentofu/pull/1728/commits/fa3d716b993c49a1f814dd35d2dfce6a80741b51)